ts/screens/wallet/payment/ScanQRCodeScreen.tsx

```diff
@@ -106,14 +127,36 @@ const styles = StyleSheet.create({
   }
 });
 
-export class QRcodeAcquisitionByScannerScreen extends React.Component<
-  Props,
-  never
-> {
+const rptIdFromQrCodeString = (
+  qrCodeString: string
+): Either<t.Errors, RptId> => {
+  const qrCode = PaymentNoticeQrCodeFromString.decode(qrCodeString);
+  if (qrCode.isRight()) {
+    // the return value is either Left or Right
+    // (should be Right if PaymentNoticeQrCodeFromString is successful)
+    return rptIdFromPaymentNoticeQrCode(qrCode.value);
+  } else {
+    return qrCode; // is Left already
+  }
+};
+
+class ScanQRCodeScreen extends React.Component<Props, never> {
   private goBack() {
     this.props.navigation.goBack();
   }
 
+  private qrCodeRead = (data: string) => {
+    const rptId = rptIdFromQrCodeString(data);
+    const qrCode = PaymentNoticeQrCodeFromString.decode(data);
```

Comment:
`qrCode` -> `paymentNotice` ?

Reply:
why two different functions for processing one qrcode string?

Reply:
Because `RptId` does not contain the information about the amount of the transaction, while `PaymentNotice` does: so `RptId` is used for paying (i.e. for the "verifica"), while `PaymentNotice` is used to extract the amount and compare it with the one returned by the "verifica"

Reply:
what I meant was, why not having one single function that returns a tuple?

Reply:
Because those are italia-ts-commons functions, so it probably makes sense not to tailor them to the exact needs of the app, making them scarcely reusable in the process.